
fix -Wmissing-field-initializers compiler warnings
and use compiler option for future compile checks

Please double check default field values

Signed-off-by: Andreas Mohr <and@gmx.li>

--- a/m4.include/mc-cflags.m4
+++ b/m4.include/mc-cflags.m4
@@ -55,6 +55,7 @@
     MC_CHECK_ONE_CFLAG([-Wmaybe-uninitialized])
     MC_CHECK_ONE_CFLAG([-Wmissing-braces])
     MC_CHECK_ONE_CFLAG([-Wmissing-declarations])
+    MC_CHECK_ONE_CFLAG([-Wmissing-field-initializers])
     MC_CHECK_ONE_CFLAG([-Wmissing-parameter-type])
     MC_CHECK_ONE_CFLAG([-Wmissing-prototypes])
     MC_CHECK_ONE_CFLAG([-Wnested-externs])
--- a/src/filemanager/file.c
+++ b/src/filemanager/file.c
@@ -599,7 +599,7 @@
 progress_update_one (file_op_total_context_t * tctx, file_op_context_t * ctx, off_t add)
 {
     struct timeval tv_current;
-    static struct timeval tv_start = { };
+    static struct timeval tv_start = { 0 };
 
     tctx->progress_count++;
     tctx->progress_bytes += (uintmax_t) add;
--- a/src/filemanager/find.c
+++ b/src/filemanager/find.c
@@ -199,7 +199,9 @@
 
 static find_file_options_t options = {
     TRUE, TRUE, TRUE, FALSE, FALSE,
-    FALSE, TRUE, FALSE, FALSE, FALSE, FALSE
+    FALSE, TRUE, FALSE, FALSE, FALSE, FALSE,
+    FALSE,
+    NULL
 };
 
 static char *in_start_dir = INPUT_LAST_TEXT;
--- a/src/keybind-defaults.c
+++ b/src/keybind-defaults.c
@@ -272,12 +272,12 @@
 
 /* tree */
 static const global_keymap_ini_t default_tree_keymap[] = {
-    {"Help = f1"},
-    {"Rescan = f2; ctrl-r"},
-    {"Forget = f3"},
-    {"ToggleNavigation = f4"},
-    {"Copy = f5"},
-    {"Move = f6"},
+    {"Help", "f1"},
+    {"Rescan", "f2; ctrl-r"},
+    {"Forget", "f3"},
+    {"ToggleNavigation", "f4"},
+    {"Copy", "f5"},
+    {"Move", "f6"},
 #if 0
     {"MakeDir", "f7"},
 #endif
